src/topology/instances/add_circle.lean

```diff
@@ -279,3 +311,101 @@ local attribute [instance] fact_zero_lt_one
+
+section linear_ordered_add_comm_group
+
+variables [linear_ordered_add_comm_group 𝕜] [topological_space 𝕜] [order_topology 𝕜]
+(p a : 𝕜) [hp : fact (0 < p)]
+
+include hp
+
+local notation `𝕋` := add_circle p
+
+/-- The relation identifying the endpoints of `Icc a (a + p)`. -/
+inductive endpoint_ident : Icc a (a + p) → Icc a (a + p) → Prop
+| mk : endpoint_ident
+    ⟨a,      left_mem_Icc.mpr $ le_add_of_nonneg_right hp.out.le⟩
+    ⟨a + p, right_mem_Icc.mpr $ le_add_of_nonneg_right hp.out.le⟩
+
+variables [archimedean 𝕜]
+
+/-- The equivlence between `add_circle p` and the quotient of `[a, a + p]` by the relation
+identifying the endpoints. -/
+def equiv_Icc_quot : 𝕋 ≃ quot (endpoint_ident p a) :=
+{ to_fun := λ x, quot.mk _ $ subtype.map id Ico_subset_Icc_self (equiv_Ico _ _ x),
+  inv_fun := λ x, quot.lift_on x coe $ by { rintro _ _ ⟨_⟩, exact (coe_add_period p a).symm },
+  left_inv := (equiv_Ico p a).symm_apply_apply,
+  right_inv := quot.ind $ by
+  { rintro ⟨x, hx⟩,
+    have := _,
+    rcases ne_or_eq x (a + p) with h | rfl,
+    { revert x, exact this },
+    { rw ← quot.sound endpoint_ident.mk, exact this _ _ (lt_add_of_pos_right a hp.out).ne },
+    intros x hx h,
+    congr, ext1,
+    apply congr_arg subtype.val ((equiv_Ico p a).right_inv ⟨x, hx.1, hx.2.lt_of_ne h⟩) } }
+
+end linear_ordered_add_comm_group
+
+section real
+
+variables {p a : ℝ} [hp : fact (0 < p)]
+include hp
+
+local notation `𝕋` := add_circle p
+
+/-- doesn't work if inlined in `homeo_of_equiv_compact_to_t2` -- why? -/
+private lemma continuous_equiv_Icc_quot_symm : continuous (equiv_Icc_quot p a).symm :=
+continuous_quot_lift _ $ (add_circle.continuous_mk' p).comp continuous_subtype_coe
+
+/-- The natural map from `[a, a + p] ⊂ ℝ` with endpoints identified to `ℝ / ℤ • p`, as a
+homeomorphism of topological spaces. -/
+def homeo_Icc_quot : 𝕋  ≃ₜ quot (endpoint_ident p a):=
+(continuous.homeo_of_equiv_compact_to_t2 continuous_equiv_Icc_quot_symm).symm
+
+/-! We now show that a continuous function on `[0, 1]` satisfying `f 0 = f 1` is the
+pullback of a continuous function on `unit_add_circle`. -/
+
+private lemma satisfies_rel {f : ℝ → B} (hf : f a = f (a + p)) (x y : Icc a (a + p)) :
+endpoint_ident p a x y → f x = f y := by { rintro ⟨_⟩, exact hf }
+
+private lemma lift_Ico_eq_lift_Icc {f : ℝ → B} (h : f a = f (a + p)) :
+lift_Ico p a f = (quot.lift (restrict (Icc a $ a + p) f) $ satisfies_rel h) ∘ equiv_Icc_quot p a :=
+funext (λ x, by refl)
+
+lemma lift_Ico_continuous [topological_space B] {f : ℝ → B}
+  (hf : f a = f (a + p)) (hc : continuous_on f $ Icc a (a + p)) :
+  continuous (lift_Ico p a f) :=
+begin
+  rw lift_Ico_eq_lift_Icc hf,
+  refine continuous.comp _ homeo_Icc_quot.continuous_to_fun,
+  rw continuous_coinduced_dom,
+  exact continuous_on_iff_continuous_restrict.mp hc,
+end
+
+end real
+
+section zero_based
+
+variables {p : ℝ} [hp : fact (0 < p)]
+include hp
+
+lemma lift_Ico_coe_apply' {f : ℝ → B} {x : ℝ} (hx : x ∈ Ico 0 p) :
```

Comment:
```suggestion
lemma lift_Ico_zero_coe_apply {f : ℝ → B} {x : ℝ} (hx : x ∈ Ico 0 p) :
```
Primes are always ambiguous, if you can disambiguate differently like there this is better. Same thing in the next lemma.